improver_tests/snow_fraction/test_SnowFraction.py

```diff
@@ -75,24 +76,36 @@ def setup_cubes(rain_data=RAIN_DATA, snow_data=SNOW_DATA, name="{phase}rate"):
         time_bounds=time_bounds,
         spatial_grid="equalarea",
         attributes=COMMON_ATTRS,
+        standard_grid_metadata="uk_ens",
     )
     return rain, snow
 
 
+@pytest.mark.parametrize("model_id_attr", (None, "mosg__model_configuration"))
+@pytest.mark.parametrize("mask_what", ("none", "rain", "snow", "rain and snow"))
 @pytest.mark.parametrize(
     "cube_name", ("{phase}rate", "thickness_of_{phase}fall_amount")
 )
-def test_basic(cube_name):
+def test_basic(cube_name, mask_what, model_id_attr):
     """Run a test with four values, including one that will trigger divide-by-zero.
-    Check data and metadata of result."""
+    Check data and metadata of result. Check with and without masked arrays
+    and with and without a model_id_attr value."""
```

Comment:
This took me a while to figure out - you create masked arrays with no masked points in them.  You probably need to explain this, since there's a test below for masked data that expects to raise an error.

Reply:
Doc-string updated.